content/en/docs/concepts/workloads/controllers/deployment.md

```diff
@@ -311,8 +311,7 @@ up to 3 replicas, as well as scaling down the old ReplicaSet to 0 replicas.
 ### Rollover (aka multiple updates in-flight)
 
 Each time a new Deployment is observed by the Deployment controller, a ReplicaSet is created to bring up
-the desired Pods. If the Deployment is updated, the existing ReplicaSet that controls Pods whose labels
-match `.spec.selector` but whose template does not match `.spec.template` are scaled down. Eventually, the new
+the desired Pods. If the Deployment is updated, the existing ReplicaSet with its label(s) in `.spec.selector` matching the pod's template label(s) in `.spec.template` are scaled up but those pod's template label(s) in `.spec.template` not matching with the ReplicaSet's labels in `.spec.selector` are scaled down. Eventually, the new
```

Comment:
Hi @MIhirMishra . I agree that this paragraph could be improved. It may help to break up the long sentence.
Would  something like this work:
```
Each time a new Deployment is observed by the Deployment controller, a ReplicaSet is created to bring up
the desired Pods. If the Deployment is updated, the existing ReplicaSet and corresponding Pods are updated as follows:
A ReplicaSet scales up if its labels in `.spec.selector` match the Pod template labels in `.spec.template`; otherwise,
the ReplicaSet scales down.
Eventually, the new ReplicaSet is scaled to match `.spec.replicas` and all prior ReplicaSets are scaled to 0.
```